src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -236,6 +236,11 @@ public override void OnActionExecuted(ActionExecutedContext context)
                 context.Result = CreateOperationOutcomeResult(Core.Resources.OperationCanceled, OperationOutcome.IssueSeverity.Error, OperationOutcome.IssueType.Timeout, HttpStatusCode.RequestTimeout);
                 context.ExceptionHandled = true;
             }
+            else if (context.Exception is ArgumentNullException argumentNullException)
+            {
+                context.Result = CreateOperationOutcomeResult(argumentNullException.Message, OperationOutcome.IssueSeverity.Error, OperationOutcome.IssueType.Invalid, HttpStatusCode.BadRequest);
+                context.ExceptionHandled = true;
+            }
```

Comment:
@mahajan-xor I do not think we can return BadRequest for ArgumentNullException always. This exception can happen not only for invalid URL parameters but for other reasons too.
And also, can you pls add test cases around the change?

Reply:
Added separate filter validation for above issue